org.lflang/src/org/lflang/cli/CliBase.java

```diff
@@ -111,9 +145,39 @@ public void doExecute(Io io, String[] args) {
     /**
      * The entrypoint of Picocli applications - the first method called when 
      * CliBase, which implements the Runnable interface, is instantiated.
-     * Lfc and Lff have their own specific implementations for this method.
      */ 
-    public abstract void run();
+    public void run() {
+        // If args are given in a json file, store its contents in jsonString.
+        if (topLevelArg.jsonFile != null) {
+            try {
+                topLevelArg.jsonString = new String(Files.readAllBytes(
+                        io.getWd().resolve(topLevelArg.jsonFile)));
+            } catch (IOException e) {
+                reporter.printFatalErrorAndExit(
+                        "No such file: " + topLevelArg.jsonFile);
+            }
+        }
+        // If args are given in a json string, (1) unpack them into an args
+        // array, and (2) call cmd.execute on them, which assigns them to their
+        // correct instance variables, then (3) recurses into run().
+        if (topLevelArg.jsonString != null) {
+            // Unpack args from json string.
+            String[] args = jsonStringToArgs(topLevelArg.jsonString);
+            // Execute application on unpacked args.
+            CommandLine cmd = spec.commandLine();
+            int exitCode = cmd.execute(args);
+            io.callSystemExit(exitCode);
+        // If args are already unpacked, invoke tool-specific logic.
+        } else {
+            runTool();
```

Comment:
```suggestion
            doRun();
```
This is just a matter of aesthetics...